Assets/Scripts/DataController.cs

```diff
@@ -1,10 +1,13 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
 using UnityEngine;
+using UnityEngine.SceneManagement;
 using System;
 
 public class DataController : MonoBehaviour
 {
+    private static DataController dataController;
+
```

Comment:
Suggest you add a public static method to get a reference to the instance, then in every piece of code that needs a DataController, you use DataController.GetInstance() to get that reference on load, instead of a FindObjectOfType operation 
```
private static DataController dataController;

    public static DataController GetInstance()
    {
        return dataController; 
    }
```
We shouldn't get null references if we have this in the loading screen.

Reply:
Good plan!  Can whack some of those `Find` and `FindObjectOfType` calls with this in place.  Fixed.